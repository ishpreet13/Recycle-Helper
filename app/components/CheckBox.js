import React, { useState } from "react";
import { StyleSheet, Pressable } from "react-native";
import { Ionicons } from "@expo/vector-icons";

function CheckBox() {
  const [checked, onChange] = useState(false);

  function onCheckmarkPress() {
    onChange(!checked);
  }
  return (
    <Pressable
      style={[styles.checkBoxBase, checked && styles.checkBoxChecked]}
      onPress={onCheckmarkPress}
    >
      {checked && <Ionicons name="checkmark" size={24} color={"white"} />}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  checkBoxBase: {
    width: 24,
    height: 24,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 4,
    borderWidth: 2,
    borderColor: "coral",
    backgroundColor: "transparent",
  },
  checkBoxChecked: {
    backgroundColor: "coral",
  },
});

export default CheckBox;
