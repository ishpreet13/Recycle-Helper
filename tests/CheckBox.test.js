import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import CheckBox from "../app/components/CheckBox";

describe("Testing CheckBox Component", () => {
  const wrapper = renderer.create(<CheckBox />);
  it("should render", () => {
    expect(wrapper.toJSON()).toBeTruthy();
  });
});

describe("<CheckBox />", () => {
  let tree;
  it("should render correctly", async () => {
    tree = renderer.create(<CheckBox />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("Testing CheckBox Component Width", () => {
  const wrapper = renderer.create(<CheckBox />);
  const styles = wrapper.toJSON().props.style;
  const width = styles[0].width;
  it("should have width as 24", () => {
    expect(width).toBe(24);
  });
});

describe("Testing CheckBox Component Style", () => {
  const wrapper = renderer.create(<CheckBox />);
  const styles = wrapper.toJSON().props.style;
  const height = styles[0].height;
  const borderRadius = styles[0].borderRadius;
  const borderWidth = styles[0].borderWidth;
  const borderColor = styles[0].borderColor;
  it("should have height as 24", () => {
    expect(height).toBe(24);
  });
  it("should have borderRadius as 4", () => {
    expect(borderRadius).toBe(4);
  });
  it("should have borderWidth as 2", () => {
    expect(borderWidth).toBe(2);
  });
  it("should have borderColor as coral", () => {
    expect(borderColor).toBe("coral");
  });
});
