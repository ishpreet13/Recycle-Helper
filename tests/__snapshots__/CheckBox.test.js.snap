// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "coral",
        "borderRadius": 4,
        "borderWidth": 2,
        "height": 24,
        "justifyContent": "center",
        "width": 24,
      },
      false,
    ]
  }
/>
`;
